### PR TITLE
docs: document merge of graph node to graphprotocol

### PR DESCRIPTION
**Pull request by suchapalaver**
## [SQL-264](https://linear.app/semiotic/issue/SQL-264/document-merge-of-graph-node-to-graphprotocol)

See [request for end-user and operator documentation](https://github.com/graphprotocol/graph-node/pull/5382#issuecomment-2089302252) in the discussion on the PR to merge the SQL Service into `graphprotocol/graph-node`.

### CODE REVIEW

File: docs/sql_service.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# SQL Service
```
Comment (anirudh2) [inline docs/sql_service.md]: Every instance of "SQL Service" should be replaced with "Subgraph:SQL service.

File: docs/sql_service.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+The SQL Service, developed by Semiotic Labs in collaboration with TheGuild
```
Comment (anirudh2) [inline docs/sql_service.md]: This sentence can be confusing. I recommend splitting the two ideas up. Something like:

"... provides a secure SQL interface for querying a subgraph's entities. To deploy this with minimal changes to the existing indexer stack, we do this by wrapping the SQL query that a consumer sends in a GraphQL query. For example, given a consumer SQL query [], we create a corresponding GraphQL query []."

Comment (anirudh2) [reply docs/sql_service.md]: Also, it's "The Guild," not "TheGuild"

File: docs/sql_service.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+Alternatively, configure the environment variable in your deployment scripts or
+environment setup as needed.
+
+### SQL Coverage
+
+The SQL Service covers a wide range of SQL functionality, allowing you to execute `SELECT`
+queries against your database. It supports basic querying, parameter binding, and result
+formatting into JSON or CSV.
```
Comment (anirudh2) [inline docs/sql_service.md]: Is the result formatting stuff something we've built into graph-node?

File: docs/sql_service.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+- **Table Name Validation**: Validates table names referenced in SQL queries, identifying
+unknown tables and ensuring compatibility with the schema.
+- **Common Table Expression (CTE) Handling**: Handles common table expressions, adding them
+to the set of known tables during validation.
+
+See the test suite in `store/postgres/src/sql/validation.rs` for examples of various scenarios
+and edge cases encountered during SQL query validation, including function whitelisting and
+blacklisting, multi-statement queries, unknown table references, and more.
+
+## Querying with SQL Service
```
Comment (anirudh2) [inline docs/sql_service.md]: This documentation is meant for indexers, right? I'd move this whole section up to give context to some of the above sections and change the language a bit such that instead of saying "you (the indexer)," this reads "a consumer."

**Review by anirudh2 — CHANGES_REQUESTED** [review]

File: docs/sql_service.md (hunk 3 above)
Comment (suchapalaver) [reply docs/sql_service.md]: Yes, I believe output formatting in either CSV or JSON is a sub-feature of the Subgraph:SQL Service.

File: docs/sql_service.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
+# Subgraph:SQL Service
+
+The Subgraph:SQL Service, developed by Semiotic Labs in collaboration with The Guild
+and Edge & Node, offers a secure SQL interface for querying a subgraph's entities.
+To deploy this with minimal changes to the existing indexer stack, we wrap a consumer's
```
Comment (anirudh2) [inline docs/sql_service.md]: Oh. Sorry. Instead of "we," which is ambiguous (an indexer might ask if they have to do this), you can say something like "When a consumer sends a SQL query, they (or the Studio they use), must wrap it in a GraphQL query. This is explained in the next section."

**Review by anirudh2 — CHANGES_REQUESTED** [review]
Small thing. Once you do this, I'll be good to approve!

**Review by anirudh2 — APPROVED** [review]

**Comment by suchapalaver** [comment]
Closing this but cherry-picking the commit to [here](https://github.com/semiotic-ai/graph-node/pull/29).
